Libraries/Opc.Ua.Client/Session/Session.cs

```diff
@@ -3652,6 +3745,279 @@ public bool RemoveTransferredSubscription(Subscription subscription)
+                    }
+
+                    // split input into batches
+                    int batchCount = 0;                    
+
+                    List<NodeId> nodesToBrowseForNextPass = new List<NodeId>();
+                    List<ReferenceDescriptionCollection> referenceDescriptionsForNextPass = new List<ReferenceDescriptionCollection>();
+                    List<ServiceResult> errorsForNextPass = new List<ServiceResult>();
+
+                    // loop over the batches
+                    foreach ( var nodesToBrowseBatch in ( (List<NodeId>)nodesToBrowseForPass ).Batch<NodeId, List<NodeId>>( maxNodesPerBrowse ))
+                    {
+                        int batchOffset = batchCount * (int)maxNodesPerBrowse;
+
+                        BrowseWithBrowseNext(
+                                view,
+                                nodesToBrowseBatch,
+                                maxResultsToReturn,
+                                browseDirection,
+                                referenceTypeId,
+                                includeSubtypes,
+                                nodeClassMask,
+                                out var resultForBatch,
+                                out var errorsForBatch
+                            );
+
+                        for (int ii = 0; ii < nodesToBrowseBatch.Count; ii++)
+                        {
+                            if(errorsForBatch[ii].StatusCode == StatusCodes.BadNoContinuationPoints ||
+                                errorsForBatch[ii].StatusCode == StatusCodes.BadContinuationPointInvalid)
+                            {
+                                nodesToBrowseForNextPass.Add( nodesToBrowseForPass[batchOffset + ii] );                                
+                                referenceDescriptionsForNextPass.Add( resultForPass[batchOffset + ii] );                                
+                                errorsForNextPass.Add( errorsForPass[batchOffset + ii] );
+                            }
+
+                            resultForPass[batchOffset + ii].Clear();
+                            resultForPass[batchOffset + ii].AddRange( resultForBatch[ii] );
+                            errorsForPass[batchOffset + ii] = errorsForBatch[ii];
+                            
+                        }
+                        int badNoCp = errorsForBatch.Count(x => x.StatusCode == StatusCodes.BadNoContinuationPoints);
+                        int badCpI = errorsForBatch.Count(x => x.StatusCode== StatusCodes.BadContinuationPointInvalid);
+                        int bad = errorsForBatch.Count(x => StatusCode.IsBad(x.StatusCode));
+                        badNoCPErrorsPerPass += badNoCp;
+                        badCPInvalidErrorsPerPass += badCpI;
+                        otherErrorsPerPass += bad - badNoCp - badCpI;
+
+                        batchCount++;
+                    }
+
+                    resultForPass = new List<ReferenceDescriptionCollection>();
```

Comment:
is it more efficient to just assign the referenceDescriptionsForNextPass to resultForPass and then assign referenceDescriptionsForNextPass with a new List?

Reply:
it looks at least more efficient and removes three lines of code. Will update in next commit.